RealmSwift/Sync.swift

```diff
@@ -228,6 +236,58 @@ public typealias Provider = RLMIdentityProvider
      */
     internal let stopPolicy: RLMSyncStopPolicy
 
+    /**
+     An enum which determines file recovery behvaior in the event of a client reset.
+     - note: Defaults to `.manual`
+
+     - see: `RLMClientResetMode`
+     - see: https://docs.mongodb.com/realm/sync/error-handling/client-resets/
+    */
+    public let clientResetMode: ClientResetMode
+
+    /**
+     A callback which notifies prior to  prior to a client reset occurring.
+     The Realm argument contains the local database state prior to client reset.
+     ```
+     configuration.syncConfiguration?.notifyBeforeClientReset = { localRealm in
+        let results = localRealm.objects(MyClass.self)
+        ...
```

Comment:
`...` is carrying a lot of weight here. It's not clear what if anything anyone would actually write here.